apps/astarte_realm_management/test/astarte_realm_management/device_removal/core_test.exs

```diff
@@ -0,0 +1,242 @@
+              device <- Astarte.Core.Generators.Device.device(interfaces: interface),
+              endpoint_id = Ecto.UUID.bingenerate()
+            ) do
+        keyspace = Realm.keyspace_name(realm)
+
+        %IndividualProperty{
+          device_id: device.device_id,
+          interface_id: interface.interface_id,
+          endpoint_id: endpoint_id,
+          path: "/example/path"
+        }
+        |> Repo.insert!(prefix: keyspace)
+
+        Core.delete_individual_properties!(realm, device.device_id)
+
+        assert Queries.retrieve_individual_properties_keys!(realm, device.device_id) == []
+      end
+    end
+
+    property "delete_object_datastream/2 removes object datastreams of a valid device", %{
+      realm: realm
+    } do
+      keyspace = Realm.keyspace_name(realm)
+
+      check all(
+              interface <-
+                Astarte.Core.Generators.Interface.interface(
+                  type: :datastream,
+                  aggregation: :object,
+                  explicit_timestamp: true
+                ),
+              device <- Astarte.Core.Generators.Device.device(interfaces: interface),
+              value_timestamp = DateTime.utc_now(),
+              reception_timestamp = DateTime.utc_now(),
+              reception_timestamp_submillis <- integer(0..10)
+            ) do
+        interface_json = Jason.encode!(interface)
+        _ = Engine.install_interface(realm, interface_json)
+
+        table_name =
+          CQLUtils.interface_name_to_table_name(interface.name, interface.major_version)
+
+        """
+        INSERT INTO #{keyspace}.#{table_name}
+        (device_id, path, value_timestamp, reception_timestamp, reception_timestamp_submillis)
+        VALUES (?, ?, ?, ?, ?)
+        """
+        |> Repo.query!([
+          device.device_id,
+          "/example/path",
+          value_timestamp,
+          reception_timestamp,
+          reception_timestamp_submillis
+        ])
+
+        %Device{
+          device_id: device.device_id,
+          introspection: %{interface.name => interface.major_version}
+        }
+        |> Repo.insert!(prefix: keyspace)
+
+        Core.delete_object_datastream!(realm, device.device_id)
+
+        assert Queries.retrieve_object_datastream_keys!(realm, device.device_id, table_name) == []
+
+        _ = Queries.delete_interface(realm, interface.name, interface.major_version)
+      end
+    end
+
+    property "delete_aliases/2 removes all aliases of a valid device", %{
+      realm: realm
+    } do
+      keyspace = Realm.keyspace_name(realm)
+
+      check all(
+              device_alias <- string(:utf8, length: 1..100),
+              object_type <- integer(0..256),
+              device <- Astarte.Core.Generators.Device.device(interfaces: [])
+            ) do
+        %Name{
+          object_uuid: device.device_id,
+          object_name: device_alias,
+          object_type: object_type
+        }
```

Comment:
Consider moving this to a generator (even just a private one with `defp`) so that the test is clearer, as agreed upon IRL, It also holds for other data to be seeded